### PR TITLE
COST-1446: Update Cost Model API for Cost Distribution

### PR DESCRIPTION
**Pull request by myersCody**
Criteria:
- Only allow the distribution choice when doing a monthly node or a monthly cluster cost
- Make sure the distribution type is either memory or cpu
- Update openapi.json

You can run `make load-test-customer-data` to test. 
Then you will need to delete the default cost model and put in the new one:
```
{
    "name": "Cost Management OpenShift Cost Model",
    "description": "A cost model of on-premises OpenShift clusters.",
    "source_type": "OCP",
    "source_uuids": ["e7308aa0-961b-46f5-8a3a-b31c926b387e"],
    "rates": [{
        "metric": {"name": "cpu_core_usage_per_hour"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 0.0070000000,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Supplementary"
    },
    {
        "metric": {"name": "cpu_core_request_per_hour"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 0.2000000000,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Supplementary"
    },
    {
        "metric": {"name": "memory_gb_usage_per_hour"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 0.0090000000,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Supplementary"
    },
    {
        "metric": {"name": "memory_gb_request_per_hour"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 0.0500000000,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Supplementary"
    },
    {
        "metric": {"name": "storage_gb_usage_per_month"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 0.01,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Supplementary"
    },
    {
        "metric": {"name": "storage_gb_request_per_month"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 0.01,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Supplementary"
    },
    {
        "metric": {"name": "node_cost_per_month"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 1000.0,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Infrastructure"
    },
    {
        "metric": {"name": "cluster_cost_per_month"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 1000.0,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Infrastructure"
    }
    ],
    "distribution": "cpu"
}
```

### CODE REVIEW

**Comment by myersCody** [comment]
I think we may want to considering merging this into a feature branch instead of master. Cost distribution does have the possibility to make the openshift cost totals wonky so it may be better to test in a feature branch and merging one big feature branch in the end.

**Comment by myersCody** [comment]
After a meeting with UI/UX today, we decided to move the cost distribution type to the top level of the cost model instead of rate based. I will need to update this pr to reflect those new preferences; therefore, I am moving the pr back to WIP until I can get the updates pushed.

File: koku/api/metrics/constants.py [hunk 1]
```diff
@@ -36,13 +38,21 @@
...
 SOURCE_TYPE_MAP = {
     Provider.PROVIDER_OCP: "OpenShift Container Platform",
     Provider.PROVIDER_AWS: "Amazon Web Services",
     Provider.PROVIDER_AZURE: "Microsoft Azure",
     Provider.PROVIDER_GCP: "Google Cloud Platform",
 }
 
+DISTRIBUTION_MAP = {
+    OCP_NODE_MONTH: [MEMORY_DISTRIBUTION, CPU_DISTRIBUTION],
+    OCP_CLUSTER_MONTH: [MEMORY_DISTRIBUTION, CPU_DISTRIBUTION],
```
Comment (dccurtis) [inline koku/api/metrics/constants.py:53]: Need `OCP_PVC_MONTH` as well?

**Review by dccurtis — DISMISSED** [review]
Looks good 👍 

We were talking about having QE review API change PRs.  I added @project-koku/quality-engineers as a reviewer.

Some ideas for IQE enhancements are:
* Verify that cpu distribution type is in the cost model response since it's default
* Add a crud test that changes to memory and back to cpu.  Verify response but don't expect it to actually change any calculations yet.

**Review by dchorvat1 — APPROVED** [review]

**Pull request merged**